packages/flutter_tools/test/general.shard/web/proxy_test.dart

```diff
@@ -0,0 +1,449 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/isolated/proxy_middleware.dart';
+import 'package:flutter_tools/src/web/devfs_proxy.dart';
+import 'package:shelf/shelf.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+import '../../src/testbed.dart';
+
+void main() {
+  late TestBed testbed;
+  setUp(() {
+    testbed = TestBed();
+  });
+
+  group('ProxyRule.fromYaml', () {
+    test(
+      'should create SourceProxyRule with source and no replacement',
+      () => testbed.run(() {
+        final yaml =
+            loadYaml('''
```

Comment:
```suggestion
            loadYaml(r'''
```

? Some strings in the test are raw, others aren't. Please, make them all consistent!